### PR TITLE
Add basic PDA/Syndicate Uplink.

### PR DESCRIPTION
**Pull request by FL-OZ**
Revives: #915

### CODE REVIEW

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        /// <param name="acc"></param>
+        public void InitUplinkAccount(UplinkAccount acc)
+        {
+            _syndicateUplinkAccount = acc;
+            _uplinkManager.AddNewAccount(_syndicateUplinkAccount);
+
+            _syndicateUplinkAccount.BalanceChanged += account =>
+            {
+                _syndicateUplinkAccount = account;
+                var accData = new UplinkAccountData(_syndicateUplinkAccount.AccountHolder, _syndicateUplinkAccount.Balance);
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/PDA/PDAComponent.cs:199]: Won't `UpdatePDAUserInterface` do this?

File: Content.Server/GameObjects/Components/PDA/PDAComponent.cs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                _interface.SetState(new PDAUpdateUplinkAccountMessage(accData));
+                var listingsMessage = new PDASendUplinkListingsMessage(_uplinkManager.FetchListings().ToArray());
+                _interface.SetState(listingsMessage);
+                UpdatePDAUserInterface();
+            };
+
+            var accData = new UplinkAccountData(_syndicateUplinkAccount.AccountHolder, _syndicateUplinkAccount.Balance);
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/PDA/PDAComponent.cs:206]: Same question here.

File: Content.Server/PDA/PDAUplinkManager.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IPrototypeManager _prototypeManager;
+        [Dependency] private readonly IEntityManager _entityManager;
+#pragma warning restore 649
+
+        private List<UplinkAccount> _accounts;
+        private List<UplinkListingData> _listings;
+
+        public IReadOnlyList<UplinkListingData> FetchListings() //Shouldn't this be a lambda function or something? I did this when I was tired.
```
Comment (ComicIronic) [inline Content.Server/PDA/PDAUplinkManager.cs:26]: You could (and probably should, given your comment), make this a property.

File: Content.Shared/GameObjects/Components/PDA/UplinkCategory.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+namespace Content.Shared.GameObjects.Components.PDA
+{
+    public enum UplinkCategory
+    {
+        Weapon,
```
Comment (ComicIronic) [inline Content.Shared/GameObjects/Components/PDA/UplinkCategory.cs:5]: Should probably be controlled by some kind of prototype (is what PJB will comment).

**Comment by ComicIronic** [comment]
Cool stuff.

File: Content.Shared/GameObjects/Components/PDA/UplinkCategory.cs (hunk 4 above)
Comment (PJB3005) [reply Content.Shared/GameObjects/Components/PDA/UplinkCategory.cs:5]: I'm fine with either, honestly.

**Pull request merged**